integration/src/integration/java/pl/allegro/tech/hermes/integration/PublishingAvroTest.java

```diff
@@ -233,6 +233,22 @@ public void shouldReturnServerInternalErrorResponseOnMissingSchemaAtSpecifiedVer
         assertThat(response.getStatus()).isEqualTo(INTERNAL_SERVER_ERROR.getStatusCode());
     }
 
+    @Test
+    public void shouldReturnBadRequestResponseOnMissingMetadataFieldInSchema() {
+        // given
+        Schema schema = AvroUserSchemaLoader.load("/schema/user_no_metadata.avsc");
+        Topic topic = randomTopic("pl.allegro.test", "Topic")
+                .withContentType(AVRO)
+                .build();
+        operations.buildTopicWithSchema(topicWithSchema(topic, schema.toString()));
+
+        // when
+        Response response = publisher.publish(topic.getQualifiedName(), user.asJson());
+
+        // then
+        assertThat(response).hasStatus(BAD_REQUEST);
```

Comment:
The test should check for existence of a metadata-related user-friendly message.

Reply:
Done